jaeger-zipkin/src/main/java/io/jaegertracing/senders/zipkin/V2SpanConverter.java

```diff
@@ -0,0 +1,193 @@
+      return zipkin2.Span.Kind.CONSUMER;
+    } else if (kind.equals(Tags.SPAN_KIND_PRODUCER)) {
+      return zipkin2.Span.Kind.PRODUCER;
+    } else {
+      return null;
+    }
+  }
+
+  private static void buildAnnotations(Span span, zipkin2.Span.Builder builder) {
+    if (ConverterUtil.isRpc(span)) {
+      String startLabel = zipkin.Constants.SERVER_RECV;
+      String endLabel = zipkin.Constants.SERVER_SEND;
+      if (ConverterUtil.isRpcClient(span)) {
+        startLabel = zipkin.Constants.CLIENT_SEND;
+        endLabel = zipkin.Constants.CLIENT_RECV;
+      }
+
+      builder.addAnnotation(span.getStart(), startLabel);
+      builder.addAnnotation(span.getStart() + span.getDuration(), endLabel);
+    }
+
+    List<LogData> logs = span.getLogs();
+    if (logs != null) {
+      for (LogData logData : logs) {
+        String logMessage = logData.getMessage();
+        Map<String, ?> logFields = logData.getFields();
+        if (logMessage != null) {
+          builder.addAnnotation(logData.getTime(), logMessage);
+        } else if (logFields != null) {
+          builder.addAnnotation(logData.getTime(), gson.toJson(logFields));
+        }
+      }
+    }
+  }
+
+  private static void buildTags(Span span, zipkin2.Span.Builder builder) {
+    Map<String, Object> tags = span.getTags();
+    boolean isRpc = ConverterUtil.isRpc(span);
+    boolean firstSpanInProcess = span.getReferences().isEmpty() || ConverterUtil.isRpcServer(span);
+
+    if (firstSpanInProcess) {
+      Map<String, ?> processTags = span.getTracer().tags();
+      // add tracer tags to first zipkin span in a process but remove "ip" tag as it is
+      // taken care of separately.
+      for (Map.Entry<String, ?> entry : processTags.entrySet()) {
+        String tagKey = entry.getKey();
+        if (!Constants.TRACER_IP_TAG_KEY.equals(tagKey)) {
+          Object tagValue = entry.getValue();
+          // add a tracer. prefix to process tags for zipkin
+          builder.putTag("tracer." + tagKey, tagValue.toString());
+        }
+      }
+    }
+
+    if (!isRpc) {
+      String componentName;
+      Object componentTag = tags.get(Tags.COMPONENT.getKey());
+      if (componentTag instanceof String) {
+        componentName = componentTag.toString();
+      } else {
+        // spans always have associated tracers, and service names
+        componentName = span.getTracer().getServiceName();
+      }
+
+      builder.putTag(zipkin.Constants.LOCAL_COMPONENT, componentName);
+    }
+
+    if (tags != null) {
+      for (Map.Entry<String, Object> entry : tags.entrySet()) {
+        String tagKey = entry.getKey();
+        // Every value is converted to string because zipkin search doesn't
+        // work well with ints, and bytes.
+        Object tagValue = entry.getValue();
+        builder.putTag(tagKey, tagValue.toString());
+      }
+    }
+  }
+
+  private static InetAddress convertIp(Integer ip) {
+    byte[] bytes = ByteBuffer.allocate(4).putInt(ip).array();
```

Comment:
Can `ip` be null here? If so, you probably need to handle that. If not, then you can change the method parameter to be the unboxed `int`